### PR TITLE
DSP-1756 | improve codemod / add eslint-plugin for codemod alternative

### PR DESCRIPTION
**Pull request by DarkPurple141**
Hi folks 👋 ,

This PR introduces a few changes:

1. There's a new `eslint-plugin` with its first rule; a codemod that transforms `@emotion` into `@compiled/react`.
2. I've also amended the existing codemod to allow the `css()` style function usage we apply in the Design System.

I've added / adjusted tests.

Any questions please shout out in the comments / reviewssss. 

Cheers, AH

### CODE REVIEW

**Review by at-nathan — DISMISSED** [review]
LGTM, please wait from another approval from @JakeLane or @MonicaOlejniczak before merge.

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (madou) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:20]: Do we need to fix this?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (madou) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:18]: Is not required*

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (madou) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:16]: nit: This error message seems loose. Could we tighten it up?

> `@emotion` should not be used, use `@compiled/react` in its place.

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (madou) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:47]: Can we delete this line?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (madou) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:76]: Can we type it here instead of relying on casts?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (madou) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:80]: Same here.

File: yarn.lock
Comment (madou) [inline yarn.lock:76]: Can you de-dupe deps?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:1]: Can you remove `unit/` in the file path. That convention is not yet used in this repository.

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:10]: Can you add a test for all three imports in a single line?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:37]: Can you add tests for `@emotion/react` too?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:43]: Can you indent this like so:

```
code: `
  /** @jsx jsx */
 import { css, jsx, ClassNames } from '@emotion/core';
`,
```

Same for anything else below

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:11]: This isn't in the recommended rules though?

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (MonicaOlejniczak) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:16]: Agree, though:

> in its place.

to

> instead.

is more common

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:39]: Can you place these before the `if` statements, so there's slightly less work to be done depending on when the check succeeds

**Review by MonicaOlejniczak — CHANGES_REQUESTED** [review]

File: packages/eslint-plugin/src/rules/__tests__/utils/_tester.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/__tests__/utils/_tester.tsx:1]: yes, this is just from a ds templte will delete

Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/__tests__/utils/_tester.tsx:1]: sure

File: packages/eslint-plugin/src/rules/__tests__/utils/_tester.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/__tests__/utils/_tester.tsx:4]: It's a complaint that you can't augment the type signature.

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:1]: yep

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:20]: yes

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx:43]: yes

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:11]: removed this line

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:47]: deleted

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:76]: ok we can by asserting the type further up, have now done this.

File: packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx:39]: sure.

**Review by DarkPurple141 — COMMENTED** [review]
Will re-push shortly

File: packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx
Comment (DarkPurple141) [reply packages/eslint-plugin/src/rules/emotion-to-compiled/__tests__/unit/rule.test.tsx]: yep added now

**Review by JakeLane — APPROVED** [review]

**Review by MonicaOlejniczak — APPROVED** [review]

**Comment by DarkPurple141** [comment]
I can't merge this @MonicaOlejniczak or @JakeLane over to you.

**Pull request merged**
